Makefile

```diff
@@ -174,13 +174,21 @@ package-software:  ## to package the binary
 .PHONY: docker-build
 docker-build: ## to build the docker image
 	@$(INFO) Performing Docker build ${APP_NAME}:${APP_VERSION_NO_V}
-	$(AT)$(DOCKER) build \
-	--build-arg GO_IMAGE=${DOCKER_IMAGE_GO} \
-	--build-arg=ARCH=amd64 \
+	$(AT)$(DOCKER) buildx build \
+	--no-cache --pull --platform linux/amd64,linux/arm64 \
 	-f ${DOCKER_FILE} . \
 	-t ${APP_NAME}:${APP_VERSION_NO_V} || ${FAIL}
 	@$(OK) Performing Docker build ${APP_NAME}:${APP_VERSION_NO_V}
 
+.PHONY: release-docker-build
+release-docker-build: docker-login
+	@$(INFO) Performing Release Docker build ${APP_NAME}:${APP_VERSION_NO_V}
+	$(AT)$(DOCKER) buildx build \
+	--no-cache --pull --platform linux/amd64,linux/arm64 \
+	-f ${DOCKER_FILE} . \
+	-t $(DOCKER_REGISTRY)/${DOCKER_REGISTRY_REPO}:${APP_VERSION_NO_V} --push || ${FAIL}
+	@$(OK) Performing Release Docker build ${APP_NAME}:${APP_VERSION_NO_V}
+
```

Comment:
If I'm correct, the `release-docker-build` target, will replace the `docker-push`  target. 
If that is indeed the case, let's remove the `docker-push` target from the makefile. 

Morover we have logic in place, to push latest tag also conditionally 
https://github.com/mattermost/mattermost-push-proxy/blob/master/Makefile#L189-L194 . 

https://hub.docker.com/layers/mattermost/mattermost-push-proxy/latest/images/sha256-efbb8ce1bb166a2842d8be1f3f42432c84cf6b5c8c82baad01d1e9dc580a5372?context=explore

Is that tackled here? 


Finally, let's ensure we include comments in the target. These comments are automatically parsed to generate the help documentation
https://github.com/mattermost/mattermost-push-proxy/blob/master/Makefile#L135-L139

```
 make help                                                                                                                                                                                                                         
Usage:
make help                           to get help
make build                          to build
make release                        to build and release artifacts
make package                        to build, package
make sign                           to sign the artifact and perform verification
make lint                           to lint
make test                           to test
make package-software               to package the binary
make docker-build                   to build the docker image
make docker-push                    to push the docker image
make docker-sign                    to sign the docker image
make docker-verify                  to verify the docker image
make docker-sbom                    to print a sbom report
make docker-scan                    to print a vulnerability report
make docker-lint                    to lint the Dockerfile
make docker-login                   to login to a container registry
make go-build                       to build binaries
make go-run                         to run locally for development
make go-test                        to run tests
make go-mod-check                   to check go mod files consistency
make go-lint                        to lint go code
make go-doc                         to generate documentation
make github-release                 to publish a release and relevant artifacts to GitHub
make clean                          to clean-up
```

Reply:
Reverted to docker-push target 😄